sql/src/main/java/io/crate/execution/dsl/projection/UpdateProjection.java

```diff
@@ -48,14 +53,20 @@ public UpdateProjection(Symbol uidSymbol,
                             Symbol[] assignments,
                             Version version,
                             @Nullable Symbol[] returnValues,
+                            @Nullable Field[] output,
                             @Nullable Long requiredVersion) {
         super(uidSymbol);
         this.assignmentsColumns = assignmentsColumns;
         this.assignments = assignments;
         this.allOn4_1 = version.onOrAfter(Version.V_4_1_0);
         this.returnValues = returnValues;
+        if (output != null) {
+            this.output = new Symbol[output.length];
+            for (int i = 0; i < output.length; i++) {
+                this.output[i] = (new InputColumn(i, output[i].valueType()));
```

Comment:
In most cases we put the responsibility to built the `InputColumn` symbols to who-ever is creating the projection and then add an assertion that the symbols in the arguments must not contain `Field/Reference`. See `FilterProjection` for an example.